js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorerSidebar.tsx

```diff
@@ -344,8 +350,23 @@ const Node = ({
               <Popover
                 content={
                   <Menu>
-                    {/* TODO: Hook up materialization */}
-                    <MenuItem icon="materialization" text="Materialize" />
+                    <MenuItem
+                      icon="materialization"
+                      text={
+                        <Box flex={{direction: 'row', alignItems: 'center', gap: 4}}>
+                          <span>Materialize</span>
+                          {loading ? <Spinner purpose="body-text" /> : null}
+                        </Box>
+                      }
+                      onClick={async (e) => {
+                        await showSharedToaster({
+                          intent: 'primary',
+                          message: 'Initiating materialization',
+                          icon: 'materialization',
+                        });
```

Comment:
I don't think we do this elsewhere and if we want it, we should probably move it into `useMaterializationAction` and do it in other "Materialize" menus